service/src/test/java/uk/gov/hmcts/reform/fpl/service/orders/generator/DocumentMergerTest.java

```diff
@@ -0,0 +1,92 @@
+class DocumentMergerTest {
+    @Mock
+    private DocumentConversionService documentConversionService;
+
+    @Mock
+    private DocumentDownloadService documentDownloadService;
+
+    @InjectMocks
+    private DocumentMerger underTest;
+
+    private static final String ORIGINAL_FILENAME = "document1.pdf";
+    private static final String ADDITIONAL_FILENAME = "document2.pdf";
+
+    private static final DocmosisDocument DOCMOSIS_DOCUMENT = DocmosisDocument.builder()
+        .documentTitle(ORIGINAL_FILENAME).bytes(new byte[]{1}).build();
+
+    private static final DocumentReference DOCUMENT_REFERENCE = testDocumentReference(ADDITIONAL_FILENAME);
+
+    @Test
+    void shouldMergeTheDocuments() {
+        final byte[] originalDocument = readBytes("documents/document1.pdf");
+        final byte[] additionalDocument = readBytes("documents/document2.pdf");
+        final byte[] additionalDocumentBytes = new byte[]{1, 2};
+        final byte[] mergedDocument = readBytes("documents/merged.pdf");
+
+        when(documentConversionService.convertToPdf(eq(DOCMOSIS_DOCUMENT.getBytes()), anyString()))
+            .thenReturn(originalDocument);
+
+        when(documentDownloadService.downloadDocument(eq(DOCUMENT_REFERENCE.getBinaryUrl())))
+            .thenReturn(additionalDocumentBytes);
+        when(documentConversionService.convertToPdf(eq(additionalDocumentBytes), anyString()))
+            .thenReturn(additionalDocument);
+
+        DocmosisDocument actualMergedPdf = underTest.mergeDocuments(DOCMOSIS_DOCUMENT, List.of(DOCUMENT_REFERENCE));
+
+        assertThat(actualMergedPdf.getDocumentTitle()).isEqualTo(DOCMOSIS_DOCUMENT.getDocumentTitle());
+        assertThat(actualMergedPdf.getBytes()).contains(mergedDocument);
+    }
+
+    @Test
+    void shouldMergeTheDocumentsWhenAdditionalDocumentsAreEmpty() {
+        DocmosisDocument actualMergedPdf = underTest.mergeDocuments(DOCMOSIS_DOCUMENT, List.of());
+
+        assertThat(actualMergedPdf).isEqualTo(DOCMOSIS_DOCUMENT);
+    }
+
+    @Test
+    void shouldThrowExceptionWhenAdditionalDocumentIsCorrupted() {
+        final byte[] originalDocument = readBytes("documents/document1.pdf");
+        final byte[] additionalDocumentBytes = new byte[]{1, 2};
+
+        when(documentConversionService.convertToPdf(eq(DOCMOSIS_DOCUMENT.getBytes()), anyString()))
+            .thenReturn(originalDocument);
+
+        when(documentDownloadService.downloadDocument(eq(DOCUMENT_REFERENCE.getBinaryUrl())))
+            .thenReturn(additionalDocumentBytes);
+        when(documentConversionService.convertToPdf(eq(additionalDocumentBytes), anyString()))
+            .thenReturn(additionalDocumentBytes);
+
+        Exception exception = assertThrows(DocumentMergeException.class,
+            () -> underTest.mergeDocuments(DOCMOSIS_DOCUMENT, List.of(DOCUMENT_REFERENCE))
+        );
+
+        assertThat(exception.getMessage())
+            .isEqualTo(String.format("Exception occurred while merging documents for %s",
```

Comment:
Don't think that you need the String.format here, you know the file name from the decalared fields above.